src/sql/src/parsers/error.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use datafusion_common::DataFusionError;
+use sqlparser::parser::ParserError;
+
+pub enum TQLError {
```

Comment:
Let's add some docs to this enum and its variants.

Reply:
Sure @killme2008 
I'm trying to apply Ruihang's suggestion about `snafu`. 🤞 will be ready tomorrow!